### PR TITLE
Add debouncing mechanism to avoid restarting clangd too many times

### LINKED ISSUE #98: Error restarting server when using onConfigChanged
When using onConfigChanged and the compile_commands.json file gets updated I get an error when the extension tries to restart the server

image

**System information**
Clangd version: trunk
clangd extension version: 0.18
Operating system: Ubuntu 20.04

### PR DESCRIPTION
**Pull request by rapgenic**
Fixes #98

### CODE REVIEW

**Comment by njames93** [comment]
@Trass3r Do you want to give this a try, It fixed it on my end but there could be something subtly different on your end.

File: src/config-file-watcher.ts [hunk 1]
```diff
@@ -11,6 +13,7 @@ export function activate(context: ClangdContext) {
 
 class ConfigFileWatcher {
   private databaseWatcher: vscode.FileSystemWatcher = undefined;
+  private pendingRestartTimer: NodeJS.Timer = undefined;
```
Comment (hokein) [inline src/config-file-watcher.ts:16]: maybe name it `debounceTimer`?

File: src/config-file-watcher.ts [hunk 2]
```diff
@@ -26,12 +29,23 @@ class ConfigFileWatcher {
...
 
+  async debouncedHandleConfigFilesChanged(uri: vscode.Uri) {
+    if (this.pendingRestartTimer) {
+      clearTimeout(this.pendingRestartTimer);
+    }
+
+    this.pendingRestartTimer = setTimeout(async () => {
+      await this.handleConfigFilesChanged(uri);
+      this.pendingRestartTimer = undefined;
+    }, RESTART_DEBOUNCE);
```
Comment (hokein) [inline src/config-file-watcher.ts:46]: nit: just inline the value here since we use it only once.

**Comment by rapgenic** [comment]
Done! 

I also updated with the new changes on master

**Review by hokein — APPROVED** [review]
Thanks.

**Pull request merged**